plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/form.tsx

```diff
@@ -72,6 +73,17 @@ const Form = < T extends AddressFormValues | ContactFormValues >( {
 		};
 	}, [ currentFields, currentFieldConfig, currentCountry, addressType ] );
 
+	const [ isAddress2Visible, setIsAddress2Visible ] = useState(
+		objectHasProp( values, 'address_2' ) && values.address_2 !== ''
+	);
+
+	const toggleAddress2Visibility = (
```

Comment:
Consider using [`useCallback`](https://react.dev/reference/react/useCallback) here to cache this function between renders

Reply:
Thanks for suggesting this, @opr. I've just updated the PR accordingly.